presentation-compiler/src/main/dotty/tools/pc/PcInlayHintsProvider.scala

```diff
@@ -116,26 +118,69 @@ class PcInlayHintsProvider(
               InlayHintKind.Type,
             )
             .addDefinition(adjustedPos.start)
-      case ByNameParameters(byNameParams) =>
-        def adjustByNameParameterPos(pos: SourcePosition): SourcePosition =
-          val adjusted = adjustPos(pos)
-          val start = text.indexWhere(!_.isWhitespace, adjusted.start)
-          val end = text.lastIndexWhere(!_.isWhitespace, adjusted.end - 1)
+      case Parameters(isInfixFun, args) =>        
+        def isNamedParam(pos: SourcePosition): Boolean =
+          val start = text.indexWhere(!_.isWhitespace, pos.start)
+          val end = text.lastIndexWhere(!_.isWhitespace, pos.end - 1)
 
+          text.slice(start, end).contains('=')
+
+        def isBlockParam(pos: SourcePosition): Boolean =
+          val start = text.indexWhere(!_.isWhitespace, pos.start)
+          val end = text.lastIndexWhere(!_.isWhitespace, pos.end - 1)
           val startsWithBrace = text.lift(start).contains('{')
           val endsWithBrace = text.lift(end).contains('}')
 
-          if startsWithBrace && endsWithBrace then
-            adjusted.withStart(start + 1)
+          startsWithBrace && endsWithBrace
+
+        def adjustBlockParamPos(pos: SourcePosition): SourcePosition =
+          if isBlockParam(pos) then
+            pos.withStart(pos.start + 1)
           else
-            adjusted
+            pos
+
+        val adjustedArgs = args.map {
+          case (name, pos, isByName) => (name, adjustPos(pos), isByName)
+        }
+
+        val namedParams = 
+          if params.namedParameters() && !isInfixFun then
+            adjustedArgs.collect {
+              // We don't want to show parameter names for block parameters or named parameters
+              case (name, pos, _) if !isBlockParam(pos) && !isNamedParam(pos) => (name, pos)
+            }
+          else Nil
 
-        byNameParams.foldLeft(inlayHints) {
-          case (ih, pos) => 
-            val adjusted = adjustByNameParameterPos(pos)
+        val byNameParams = 
+          if params.byNameParameters() then
+            adjustedArgs.collect {
+              case (name, pos, isByName) if isByName => (name, pos)
+            }
+          else Nil
+
+        val namedAndByNameInlayHints = 
+          namedParams.collect { 
+            case param@(name, pos) if byNameParams.contains(param) =>
+              (name.toString() + " = => ", adjustBlockParamPos(pos))
+          }
+        
+        val namedInlayHints = 
+          namedParams.collect { 
+            case param@(name, pos) if !byNameParams.contains(param) =>
+              (name.toString() + " = ", pos)
+          }
```

Comment:
```suggestion
        val namedAndByNameInlayHints = 
          namedParams.collect { 
            case param@(name, pos) if byNameParams.contains(param) =>
              (name.toString() + " = => ", adjustBlockParamPos(pos))
            case param@(name, pos) =>
              (name.toString() + " = ", pos)
          }
```
would amount to the same, no?

I would try to maybe even go over adjustedArgs and check on each case whether we want to show "name =" or "=>" or "= =>" which would mean going over the collection only once. Might be needless optimization, but I worry that in large files it might make a difference

Reply:
I agree, we will reduce the constant which with large files can slow down. Changed.